container_main.py

```
@@ -150,11 +150,21 @@ def main():
+                    # Check exit code
+                    exitcode = process.exitcode
+                    if process.name == "scheduler" and exitcode == 2:
+                        logger.critical(
+                            "Scheduler exited due to authentication lock. API server will remain running."
+                        )
+                        # Remove from monitored processes so we don't keep logging error
+                        processes.remove(process)
+                        continue
+
+                    logger.error(
+                        f"Process {process.name} died unexpectedly (exit code: {exitcode})"
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)